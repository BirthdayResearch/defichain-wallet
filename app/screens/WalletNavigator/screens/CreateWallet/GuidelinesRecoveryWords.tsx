import React from 'react'
import { ThemedScrollView, ThemedText } from '../../../../components/themed'
import { tailwind } from '../../../../tailwind'
import { translate } from '../../../../translations'

export function GuidelinesRecoveryWords (): JSX.Element {
  return (
    <ThemedScrollView
      light={tailwind('bg-white')} dark={tailwind('bg-gray-900')}
      style={tailwind('flex-1 p-4 pt-6')}
    >
      <ThemedText
        style={tailwind('text-lg font-semibold')}
      >{translate('screens/GuidelinesRecoveryWords', 'What are recovery words?')}
      </ThemedText>
      <ThemedText
        style={tailwind('mt-2')}
      >{translate('screens/GuidelinesRecoveryWords', 'Your unique 24 recovery words is a human-readable representation of your wallet private key, generated from a list of 2048 words in the BIP-39 standard.')}
      </ThemedText>
      <ThemedText
        style={tailwind('mt-5')}
      >{translate('screens/GuidelinesRecoveryWords', 'You will need your recovery words to restore your wallet on, for example, a new mobile device. Think of them as keys to your wallet and funds.')}
      </ThemedText>
      <ThemedText
        style={tailwind('mt-5')}
      >{translate('screens/GuidelinesRecoveryWords', 'If you lose your recovery words, you will not be able to restore your wallet, and you will lose access to your wallet and funds.')}
      </ThemedText>
      <ThemedText
        style={tailwind('mt-5 mb-12')}
      >{translate('screens/GuidelinesRecoveryWords', 'It is important that you write down your recovery words legibly and in the correct order. Store your recovery words safely and securely, away from prying eyes. You may want to think about keeping it safe from fire as well.')}
      </ThemedText>
    </ThemedScrollView>
  )
}
