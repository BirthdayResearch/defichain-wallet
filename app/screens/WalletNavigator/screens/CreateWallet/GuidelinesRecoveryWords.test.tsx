import { render } from "@testing-library/react-native";
import * as React from "react";
import { GuidelinesRecoveryWords } from "./GuidelinesRecoveryWords";

describe('recovery words guidelines', () => {
  it('should match snapshot', () => {
    const rendered = render(<GuidelinesRecoveryWords />)
    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
