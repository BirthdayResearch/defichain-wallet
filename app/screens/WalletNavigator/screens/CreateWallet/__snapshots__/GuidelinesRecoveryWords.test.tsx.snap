// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recovery words guidelines should match snapshot 1`] = `
<RCTScrollView
  style={
    Array [
      Object {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 24,
      },
      Object {
        "backgroundColor": "rgba(245, 245, 245, 1)",
      },
    ]
  }
>
  <View>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "RegularFont",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          Array [
            Object {
              "fontFamily": "SemiBoldFont",
              "fontSize": 18,
              "fontWeight": "600",
              "lineHeight": 28,
            },
            Object {
              "color": "rgba(0, 0, 0, 1)",
            },
          ],
        ]
      }
    >
      What are recovery words?
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "RegularFont",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          Array [
            Object {
              "marginTop": 8,
            },
            Object {
              "color": "rgba(0, 0, 0, 1)",
            },
          ],
        ]
      }
    >
      Your unique 24 recovery words is a human-readable representation of your wallet private key, generated from a list of 2048 words in the BIP-39 standard.
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "RegularFont",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          Array [
            Object {
              "marginTop": 20,
            },
            Object {
              "color": "rgba(0, 0, 0, 1)",
            },
          ],
        ]
      }
    >
      You will need your recovery words to restore your wallet on, for example, a new mobile device. Think of them as keys to your wallet and funds.
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "RegularFont",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          Array [
            Object {
              "marginTop": 20,
            },
            Object {
              "color": "rgba(0, 0, 0, 1)",
            },
          ],
        ]
      }
    >
      If you lose your recovery words, you will not be able to restore your wallet, and you will lose access to your wallet and funds.
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "RegularFont",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          Array [
            Object {
              "marginTop": 20,
            },
            Object {
              "color": "rgba(0, 0, 0, 1)",
            },
          ],
        ]
      }
    >
      It is important that you write down your recovery words legibly and in the correct order. Store your recovery words safely and securely, away from prying eyes. You may want to think about keeping it safe from fire as well.
    </Text>
  </View>
</RCTScrollView>
`;
